Daniel/device id sqlite

djih opened the pull request.

curtisliu commented on src/com/amplitude/api/AmplitudeClient.java:
```diff
@@ -948,8 +948,13 @@ static boolean upgradePrefs(Context context, String sourcePkgName, String target
             if (source.contains(sourcePkgName + ".deviceId")) {
-                target.putString(Constants.PREFKEY_DEVICE_ID,
-                        source.getString(sourcePkgName + ".deviceId", null));
+                String deviceId = source.getString(sourcePkgName + ".deviceId", null);
```
there needs to be an upgrade path from the current sharedPrefs to the database

djih replied:
Good catch

curtisliu commented on src/com/amplitude/api/DatabaseHelper.java:
```diff
@@ -22,10 +22,16 @@
     static DatabaseHelper instance;
     private static final String TAG = "com.amplitude.api.DatabaseHelper";
 
+    private static final String AMPLITUDE_STORE_TABLE_NAME = "amplitudestore";
```
how about just "store" or "values"? no need for Amplitude since the database is Amplitude specific

djih replied:
Good point, I like store more

curtisliu commented on src/com/amplitude/api/DatabaseHelper.java:
```diff
@@ -54,10 +61,38 @@ public void onCreate(SQLiteDatabase db) {
 
     @Override
     public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+        db.execSQL("DROP TABLE IF EXISTS " + AMPLITUDE_STORE_TABLE_NAME);
```
read about SQLiteOpenHelper onUpgrade. You'll need to do version by version upgrading, to preserve old data (we've never really had to use this method before)

djih commented:
Added upgrade path from sharedPrefs to db. Fixed db upgrade path. Added some more tests

curtisliu commented on src/com/amplitude/api/AmplitudeClient.java:
```diff
@@ -793,10 +794,19 @@ private String initializeDeviceId() {
+            return deviceId;
+        }
+
+        // see if device id stored in old sharedPrefs
```
I think there should be an upgrade path that happens where upgradePrefs happens, and the getDeviceId code should only look in the "current version" store

djih commented:
Reverted changes to upgradePrefs, added 2nd upgrade path, fixed tests/added new tests for 2nd upgrade path

curtisliu commented on src/com/amplitude/api/AmplitudeClient.java:
```diff
@@ -971,4 +969,35 @@ static boolean upgradePrefs(Context context, String sourcePkgName, String target
+        if (!TextUtils.isEmpty(deviceId)) {
+            DatabaseHelper dbHelper = DatabaseHelper.getDatabaseHelper(context);
+            dbHelper.insertOrReplaceKeyValue(DEVICE_ID_KEY, deviceId);
+        }
```
you should remove the source device id afterwards so this transfer only happens once.

djih commented:
* Removed device id from sharedPrefs after upgrade
* Changed session timeout from 15m to 5m
* Updated changelog and readme to explain new session handling

curtisliu commented:
LGTM
